Project/zepeto-modules/Assets/Zepeto Gesture Module/ZepetoScript/GestureLoader.ts

```diff
@@ -51,24 +55,66 @@ export default class GestureLoader extends ZepetoScriptBehaviour {
         newThumb.GetComponent<Thumbnail>().content = content;
 
         // Button Listener for each thumbnail
-        newThumb.GetComponent<Button>().onClick.AddListener(() => {
+        newThumb.GetComponent<Button>().onClick.AddListener(() => {  
             this.LoadAnimation(content);
         });
-        
         this.thumbnails.push(newThumb);
     }
 
     // 3. Loading Animation
-    private LoadAnimation(content: Content) {
+    private LoadAnimation(content: Content ) {
         // Verify animation load
         if (!content.IsDownloadedAnimation) {
             // If the animation has not been downloaded, download it.
             content.DownloadAnimation(() => {
                 // play animation clip
-                this._myCharacter.SetGesture(content.AnimationClip);
-            });
+                this.runAnimation(content.AnimationClip, content.Keywords)
+            });    
         } else {
-            this._myCharacter.SetGesture(content.AnimationClip);
+            this.runAnimation(content.AnimationClip, content.Keywords)
+        }
+    }
+       
+    // A function to run an animation, 
+    private runAnimation(animation: AnimationClip, type: OfficialContentType[] )
+    {        
+        //if there is another gesture running, stop the coroutine and cancel the gesture
+        if(this.gestureLoop)
+        {
+            this.StopCoroutine(this.gestureLoop)
+        }
+        this._myCharacter.CancelGesture()
+
+        // check if isLooping is true and it is not a pose
+        if(this.isLooping === true && this._checkIfGestureIsPose(type) === false)
+        {            
+            this.gestureLoop = this.StartCoroutine(this.setGestureLoop(animation))
+        }
+        //When the isLooping is false
+        else{
+            //If there is another animation running, cancel it
+            this._myCharacter.SetGesture(animation)
+        }   
+    }
+
+    private _checkIfGestureIsPose(type: OfficialContentType[])
```

Comment:
What about `_checkIfLoopable` or `_checkIfRepeatable`?

These might be better-read and logic-oriented names!

Reply:
As this function only goal is to check if the animation is a pose or not , maybe it should be something related to pose ? Something like `_gestureIsPose()`

Reply:
Yup. that's a good idea.

but still, I guess the main goal of this function is to determine whether the content type is repeatable and the only place the function is referenced is as below -

```
        // check if isLooping is true and it is not a pose
        if(this.isLooping === true && this._checkIfGestureIsPose(type) === false)
```

So, if we could write as below, it'd be better to read through the logic.

```
        // check if isLooping is true and is repeatable content type 
        if(this.isLooping && this._isRepeatableContentType(type))
```

This is solely a personal opinion so, it's up to the author whether to take the idea!

Reply:
Ah yeah, I got it. Will make the change accordingly.